From 8ee9d695b5edf49bfae2590ba3dba7045f0e4ef2 Mon Sep 17 00:00:00 2001
From: David Kahurani <redress@duck.com>
Date: Wed, 10 Sep 2025 13:17:25 +0300
Subject: [PATCH] io_uring: Remove failed request handling logic from
 io_submit_sqe

io_submit_fail_init handles any and all the requests with the
REQ_F_FAIL flag on. Request with REQ_F_FAIL consequently not get this
far

Additionally, so far into the code, io_submit_fail_init is the only method
that sets REQ_F_FAIL and in directly return into io_submit_sqe

Signed-off-by: David Kahurani <redress@duck.com>
---
 io_uring/io_uring.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/io_uring/io_uring.c b/io_uring/io_uring.c
index 93633613a165..9333e2693a81 100644
--- a/io_uring/io_uring.c
+++ b/io_uring/io_uring.c
@@ -2267,11 +2267,11 @@ static inline int io_submit_sqe(struct io_ring_ctx *ctx, struct io_kiocb *req,
 		/* last request of the link, flush it */
 		req = link->head;
 		link->head = NULL;
-		if (req->flags & (REQ_F_FORCE_ASYNC | REQ_F_FAIL))
+		if (req->flags & REQ_F_FORCE_ASYNC)
 			goto fallback;
 
 	} else if (unlikely(req->flags & (IO_REQ_LINK_FLAGS |
-					  REQ_F_FORCE_ASYNC | REQ_F_FAIL))) {
+					  REQ_F_FORCE_ASYNC))) {
 		if (req->flags & IO_REQ_LINK_FLAGS) {
 			link->head = req;
 			link->last = req;
-- 
2.50.1

